src/main/kotlin/graphql/kickstart/tools/MethodFieldResolver.kt

```diff
@@ -77,8 +87,15 @@ internal class MethodFieldResolver(
             val genericParameterType = this.getJavaMethodParameterType(index)
                 ?: throw ResolverError("Missing method type at position ${this.getJavaMethodParameterIndex(index)}, this is most likely a bug with graphql-java-tools")
 
+            val kotlinParameterType = this.getKotlinMethodParameterType(index)
+
             val isNonNull = definition.type is NonNullType
             val isOptional = this.genericType.getRawClass(genericParameterType) == Optional::class.java
+            val shouldValidate = if (kotlinParameterType != null) {
+                kotlinParameterType.findAnnotation<Valid>() != null
+            } else {
+                this.genericType.getRawClass(genericParameterType).isAnnotationPresent(Valid::class.java)
+            }
```

Comment:
https://youtrack.jetbrains.com/issue/KT-13228

I've not tested this with Java.